apps/fishing-map/features/map/map-sources.hooks.ts

```diff
@@ -0,0 +1,151 @@
+    }
+    return source
+  })
+  return sources
+}
+
+export const useSourceInStyle = (sourcesIds: SourcesHookInput) => {
+  const style = useMapStyle()
+  if (!sourcesIds || !sourcesIds.length) {
+    return false
+  }
+  const sourcesIdsList = getGeneratorSourcesIds(style, sourcesIds)
+  const sourcesLoaded = sourcesIdsList.every((source) => style?.sources?.[source] !== undefined)
+  return sourcesLoaded
+}
+
+export const useMapSourceTilesLoadedAtom = () => {
+  // Used it once in Map.tsx the listeners only once
+  const map = useMapInstance()
+  const setSourceTilesLoaded = useSetRecoilState(mapTilesAtom)
+
+  useEffect(() => {
+    if (!map) return
+
+    const onSourceDataLoading = (e: any) => {
+      const { sourceId } = e
+      if (sourceId) {
+        setSourceTilesLoaded((state) => ({ ...state, [sourceId]: false }))
+      }
+    }
+
+    const onSourceTilesLoaded = (e: any) => {
+      const { sourceId } = e
+      if (sourceId) {
+        setSourceTilesLoaded((state) => ({ ...state, [sourceId]: true }))
+      }
+    }
+    if (map) {
+      map.on('dataloading', onSourceDataLoading)
+      map.on('sourcetilesdata', onSourceTilesLoaded)
+    }
+    const detachListeners = () => {
+      map.off('dataloading', onSourceDataLoading)
+      map.off('sourcetilesdata', onSourceTilesLoaded)
+    }
+
+    return detachListeners
+  }, [map, setSourceTilesLoaded])
+}
+
+export const useMapSourceTiles = () => {
+  const sourceTilesLoaded = useRecoilValue(mapTilesAtom)
+  return sourceTilesLoaded
+}
+
+export const useMapSourceTilesLoaded = (sourcesId: SourcesHookInput) => {
+  const style = useMapStyle()
+  const sourceTilesLoaded = useMapSourceTiles()
+  const sourceInStyle = useSourceInStyle(sourcesId)
+  const sourcesIdsList = getGeneratorSourcesIds(style, sourcesId)
+  return sourceInStyle && sourcesIdsList.every((source) => sourceTilesLoaded[source])
+}
+
+export type DataviewFeature = {
+  sourceId: string
+  dataviewId: string
+  loaded: boolean
+  features: Feature[]
+  metadata: HeatmapLayerMeta
+}
+export const useMapDataviewFeatures = (
+  dataviews: UrlDataviewInstance[],
+  bounds?: MiniglobeBounds
+) => {
+  const style = useMapStyle()
+  const map = useMapInstance()
+  const sourceTilesLoaded = useMapSourceTiles()
+
+  const dataviewMetadata = useMemo(
+    () =>
+      dataviews.map((dataview) => {
+        const dataviewSource =
+          dataview.category === 'fishing' || dataview.category === 'presence'
+            ? MERGED_ACTIVITY_ANIMATED_HEATMAP_GENERATOR_ID
+            : dataview.id
+        const metadata = getSourceMetadata(style, dataviewSource)
+        return {
+          dataviewId: dataview.id,
+          metadata: metadata,
+        }
+      }),
+    [dataviews, style]
+  )
+
+  const dataviewFeatures = useMemo(() => {
+    const dataviewFeature = dataviewMetadata.map(({ dataviewId, metadata }) => {
+      const sourceId = metadata?.timeChunks?.activeSourceId
```

Comment:
As you can see here https://github.com/GlobalFishingWatch/frontend/pull/1210/files#diff-2865ae427da9befe4e55a82d2be9f24a7a506a01c5209013c94fd974056aaac1L65
the timebar used to check all time chunks, not only the active one.
If we only consider the active chunk, we won't see the entirety of the graph in the timebar.